### ISSUE TITLE
ERROR: No parser class available for format: png

### ISSUE THREAD

**Issue by mhabsaoui** [NatLibFi/Skosmos#711]
## At which URLs did you encounter the problem?

https://localhost:8080/skosmos/rest/v1/search?query=ab*&vocab=&lang=fr&labellang=fr
https://localhost:8080/skosmos/rest/v1/types?lang=fr

## What steps will reproduce the problem?
1. being on the main vocabularies page
2. write search term (existing or not !)
3. click search button

## What is the expected output? 
Usual search Results (found or not !).

## What do you see instead?
![capture du 2018-02-15 17-16-29](https://user-images.githubusercontent.com/18755135/36267378-fe11522e-1273-11e8-862f-b1556aba024f.png)

I got this http status code on browser console : **500 Internal Server Error**

We have no clue where the hell comes this "png" thing from => only text parsing usually, not media files actually at all...

hint:   
![capture du 2018-02-15 13-06-51](https://user-images.githubusercontent.com/18755135/36256202-311c18f6-1252-11e8-996d-1b76c31ac306.png)

## What browser did you use? (eg. Firefox, Chrome, Safari, Internet explorer)
Firefox, Chrome

Thanks.

**Comment by osma** [comment]
Can you please check the Apache error log? It probably has more details...

Also, config.inc and vocabularies.ttl content would probably be necessary

**Comment by mhabsaoui** [comment]
Ok checked the apache2 logs in "/var/log/apache2/error.log"

=> I got several time the "client denied by server configuration: /skosmos/RDR" message :/

> [Thu Feb 15 12:16:26.940827 2018] [authz_core:error] [pid 82] [client xxx.xxx.xx.xx:xx] AH01630: client denied by server configuration: /skosmos/RDR

**Comment by osma** [comment]
Not sure what's the cause for that, but it doesn't seem relevant for this error.
I would expect a PHP traceback.

**Comment by mhabsaoui** [comment]
Yep, I just realized it concerns old patched bugs ^^

Will look into it...

**Comment by mhabsaoui** [comment]
Does Skosmos logs any erros or backtrace ?

**Comment by mhabsaoui** [comment]
Well, it seems we gotCC license pngs images in *.RDF files

`<cc:License rdf:about="http://creativecommons.org/licenses/by/4.0/">
      <cc:requires rdf:resource="http://creativecommons.org/ns#Notice"/>
      <cc:requires rdf:resource="http://creativecommons.org/ns#Attribution"/>
      <foaf:logo rdf:resource="http://i.creativecommons.org/l/by/4.0/80x15.png"/>
      <foaf:logo rdf:resource="http://i.creativecommons.org/l/by/4.0/88x31.png"/>
      <dc:title xml:lang="en">Attribution 4.0 International</dc:title>
      <cc:permits rdf:resource="http://creativecommons.org/ns#Reproduction"/>
      <cc:permits rdf:resource="http://creativecommons.org/ns#Distribution"/>
      <cc:permits rdf:resource="http://creativecommons.org/ns#DerivativeWorks"/>
   </cc:License>`

This explains a part of this mysterious error, but on my PC I obviously have the same vocabularies, except I'm only on http (not https).

Digging forward...

**Comment by mhabsaoui** [comment]
Well, now at least, after going in private mode on any browser ==> Same behaviour on my PC or on Integration Server, e.g. we got the same error message for the two requests...

@osma Now, you said you nedded a backtrace => But it seems that EzasyRdf misses a png format class or so...  How to proceed ?

**Comment by osma** [comment]
I think the real question is, why is EasyRdf trying to parse a PNG file? It seems that Skosmos is for some reason trying to open a PNG URI. It does that for e.g. objects of skos:closeMatch and skos:exactMatch triples, but that doesn't seem to be the case here. A traceback would reveal what piece of Skosmos code causes this.

If you can't get a traceback, would it be possible for you to create a minimal reproducible example? In practice, a small SKOS data set that you can share as well as config.inc and vocabularies.ttl files that together demonstrate this bug. Right now I can't even start guessing what goes wrong as this is something that only happens on your machine with your data and configuration.

**Comment by mhabsaoui** [comment]
Hi @osma and thanks for your Full support agin 👍 

> why is EasyRdf trying to parse a PNG file?

Well, it seems that the ' https://localhost:1888/skosmos/rest/v1/types?lang=fr' is triggered by Skosmos.
This bug 'ERROR: No parser class available for format: png' is returned by EasyRdf. Is this Normal ?

> It does that for e.g. objects of skos:closeMatch and skos:exactMatch triples

I checked if any inside our *.rdf files. Nothing. (Only those License tags)

> A traceback would reveal what piece of Skosmos code causes this.

I'll see what I can do to get that in DEV mode... This bug seems to appear only when switching to https.

Thanks.

**Comment by mhabsaoui** [comment]
> I'll see what I can do to get that in DEV mode...

Well, I have tried to log any PHP errors by editing the "/etc/php5/apache2/php.ini" to enable php errors logging into  a file (/var/log/php-errors.log). But nothing is logged...

Any suggestion about PHP logging ?

Thanks

**Comment by mhabsaoui** [comment]
Well I got it solved. 

In fact, first I decided to prune/delete all images and rebuild all from scratch.
So moving to HTTPS meant we had to set the **DEFAULT_ENDPOINT** parameter in `config.inc` to _https://example.com/jena-fuseki/ds/sparql_.
So as I was thinking about easyrdf triggering Fuseki TDB (e.g. directly from the back-end server side), I tried to replace the DEFAULT_ENDPOINT with the **real** server adress (e.g. after being redirected behind the proxy...) ==> _http://example.com:3030/ds/sparql_
And this bug disappeared (no more failing request on types?) and the search worked on skosmos home page 👍 

PS: Now what I still don't understand => With that bug (i.e. DEFAULT_ENDPOINT being on https://..../sparql), how is it possible that the search still worked on any other vocabulary page ??  I mean search box's requests refer to the same DEFAULT_ENDPOINT value.
So is there anything related to the searching strategy or something ?

Thanks
